src/query.rs

```diff
@@ -150,37 +150,45 @@ pub async fn query(context: &mut Context, query_text: String) -> Result<(), Box<
             let mut maybe_request_id: Option<String> = None;
             match response {
                 Ok(resp) => {
-                    if let Some(header) = resp.headers().get("X-REQUEST-ID") {
-                        maybe_request_id = header.to_str().map_or(None, |l| Some(String::from(l)));
-                    }
-                    if let Some(header) = resp.headers().get("firebolt-update-parameters") {
-                        set_args(context, format!("set {}", header.to_str().unwrap()).as_str())?;
-                    }
-                    if let Some(header) = resp.headers().get("firebolt-remove-parameters") {
-                        unset_args(context, format!("unset {}", header.to_str().unwrap()).as_str())?;
-                    }
-                    if let Some(header) = resp.headers().get("firebolt-update-endpoint") {
-                        let header_str = header.to_str().unwrap();
-                        // Split the header at the '?' character
-                        if let Some(pos) = header_str.find('?') {
-                            // Extract base URL and query part
-                            let base_url = &header_str[..pos];
-                            let query_part = &header_str[pos+1..];
-
-                            // Update the context URL with just the base part
-                            context.args.host = base_url.to_string();
-
-                            // Process each query parameter
-                            for param in query_part.split('&') {
-                                if !param.is_empty() {
-                                    set_args(context, format!("set {};", param).as_str())?;
+                    let mut updated_url = false;
+                    for (header, value) in resp.headers() {
```

Comment:
you also need to handle Firebolt-Reset-Session header, which removes all parameters that were added with Firebolt-Update-Parameters